mm/2s2h/Rando/Logic/Logic.h

```diff
@@ -210,6 +210,12 @@ void ValidateRegionTimeOwnership(RandoRegionId regionId, RandoCheckId checkId, u
 #define CAN_GROW_BEAN_PLANT        \
     (HAS_ITEM(ITEM_MAGIC_BEANS) && \
      (CAN_PLAY_SONG(STORMS) || (HAS_BOTTLE && (CAN_ACCESS(SPRING_WATER) || CAN_ACCESS(HOT_SPRING_WATER)))))
+// Bean patches that auto-water on Day 2 (Doggy Racetrack, Deku Palace Upper)
+// Rain only occurs Day 2 from 7:00 AM to 5:30 PM (not Night 2)
+// Requires Day 2 access OR manual watering capability
+#define CAN_USE_DAY2_RAIN_BEAN                                               \
+    (HAS_ITEM(ITEM_MAGIC_BEANS) && (CLOCK_DAY2() || CAN_PLAY_SONG(STORMS) || \
+                                    (HAS_BOTTLE && (CAN_ACCESS(SPRING_WATER) || CAN_ACCESS(HOT_SPRING_WATER)))))
```

Comment:
Probably simpler to define this as CAN_GROW_BEAN_PLANT or day 2 and bean:

```suggestion
#define CAN_USE_DAY2_RAIN_BEAN (CAN_GROW_BEAN_PLANT || (HAS_ITEM(ITEM_MAGIC_BEANS) && CLOCK_DAY2()))
```